expression/builtin_control.go

```diff
@@ -231,17 +231,17 @@ func (c *caseWhenFunctionClass) getFunction(ctx sessionctx.Context, args []Expre
 		fieldTp.SetDecimal(0)
 		types.SetBinChsClnFlag(fieldTp)
 	}
-	argTps := make([]types.EvalType, 0, l)
+	argTps := make([]*types.FieldType, 0, l)
 	for i := 0; i < l-1; i += 2 {
 		if args[i], err = wrapWithIsTrue(ctx, true, args[i], false); err != nil {
 			return nil, err
 		}
-		argTps = append(argTps, types.ETInt, tp)
+		argTps = append(argTps, args[i].GetType(), fieldTp.Clone())
```

Comment:
The second argument should be FieldType of `int/bool` ?

Reply:
`args[i].GetType()` must be bool type because `WrapForIsTrue` has been called previously